pkg/controller/jenkins/plugins/plugin.go

```diff
@@ -24,6 +25,8 @@ var (
 	NamePattern = regexp.MustCompile(`^[0-9a-zA-Z-_]+$`)
 	// VersionPattern is the plugin version regex pattern
 	VersionPattern = regexp.MustCompile(`^[0-9a-zA-Z+\\.-]+$`)
+	// DownloadURLPattern is the plugin download url regex pattern
+	DownloadURLPattern = regexp.MustCompile(`https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)`)
```

Comment:
HTTP protocol should be acceptable

Reply:
The ? after s means that it can either be s after http or nothing. I will add a unit test to test https url.